### PR TITLE
Accept `--token` flag in download command

### PR DESCRIPTION
**Pull request by Ondama**
related to #484

### CODE REVIEW

**Comment by kytrinyx** [comment]
@Ondama Thank you so much for your contribution!

I completely forgot to specify in the "good first patch" issues that these features are for the next version of the client, which live in the `nextercism` branch, not in `master`. I'm so sorry! 🌷 

Would you be willing to redo this by branching off of the `nextercism` branch, and submitting the pull request back to it? It is a completely different setup, I'm afraid, using `dep` instead of `godeps`, and `spf13/cobra` instead of `urlfav/cli`.

**Comment by kytrinyx** [comment]
(I've updated all the "good first patch" issues to specify that these are off of `nextercism` rather than master.)

**Comment by Ondama** [comment]
I'll update the PR :)

**Comment by tleen** [comment]
@kytrinyx one way we have handled with the dichotomy of the master/nextercism branches in the go track is to add a [pull request template](https://github.com/exercism/go/blob/master/.github/PULL_REQUEST_TEMPLATE.md). It's not as good a preemptively guiding users towards the `nextercism` branch but it will give people a consistent run-down of the situation before submitting a PR. Do you think that could be valuable in this repo till `nextercism` becomes the master? 

Perhaps we should mention that there may be `nextercism` branches in repos somewhere in the docs. @Ondama did you use any specific element of the documentation to guide you in creating this PR? A note may be helpful there.

**Comment by kytrinyx** [comment]
> Do you think that could be valuable in this repo till nextercism becomes the master?

Yes!

> Perhaps we should mention that there may be nextercism branches in repos somewhere in the docs.

Also yes!

**Comment by tleen** [comment]
Awesome I'll try to get to this soon.

**Comment by Ondama** [comment]
@tleen @kytrinyx what do you think about updating the settings of the repo to make the branch under active development (`nextercism` for now) the default branch?

**Comment by kytrinyx** [comment]
> what do you think about updating the settings of the repo to make the branch under active development (nextercism for now) the default branch?

That's an excellent idea. @exercism/cli if you all agree I'll go ahead and make the change.

File: cmd/download.go [hunk 1]
```diff
@@ -197,6 +208,7 @@ type downloadPayload struct {
 func initDownloadCmd() {
 	downloadCmd.Flags().StringP("uuid", "u", "", "the solution UUID")
 	downloadCmd.Flags().StringP("track", "t", "", "the track ID")
+	downloadCmd.Flags().StringP("token", "k", "", "authentication token used to connect to exercism.io")
```
Comment (kytrinyx) [inline cmd/download.go]: Dang. I totally see why this is `k`, but now I'm wracking my brains trying to figure out if we should rename the whole `token` concept so we can have a consistent and unambiguous set of flags.

File: cmd/download.go [hunk 2]
```diff
@@ -29,6 +29,17 @@ Download other people's solutions by providing the UUID.
 `,
 	Args: cobra.ExactArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
+		token, err := cmd.Flags().GetString("token")
+		if err != nil {
+			return err
+		}
+		if token != "" {
+			RootCmd.SetArgs([]string{"configure", "--token", token})
```
Comment (kytrinyx) [inline cmd/download.go]: Oh, this is elegant! Very nice find.

File: cmd/download.go [hunk 3]
```diff
@@ -29,6 +29,17 @@ Download other people's solutions by providing the UUID.
...
+			err := RootCmd.Execute()
+			if err != nil {
```
Comment (kytrinyx) [inline cmd/download.go]: I don't have very strong feelings about this, but I tend to put this on one line when there's a single return value and it is err:

```
if err := RootCmd.Execute(); err != nil {
```

Comment (Ondama) [reply cmd/download.go]: makes sense. i'll update.

File: cmd/download.go (hunk 1 above)
Comment (Ondama) [reply cmd/download.go]: i liked when it was called `key`

Comment (kytrinyx) [reply cmd/download.go]: Yeah, we might need to go back to that. Meanwhile, we'll keep this and re-evaluate before we launch v2. Thank you!

**Pull request merged**